src/Components/Feed/Posts.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react'
+import { Post } from './Post.jsx'
+import axios from 'axios';
+import { WithPost } from '../../HOCs/WithPost.jsx';
+import { useDoLike } from '../apis/useDoLike.jsx';
+import { useGetPost } from '../apis/useGetPost.jsx';
+import { useDoUnLike } from '../apis/useDoUnlike.jsx';
+import { PostLoader } from '../Loaders/PostLoader.jsx';
+import { WithToken } from "../../HOCs/withToken";
+import { useGetUser } from '../apis/useGetUser.jsx';
+import { useDeletePost } from '../apis/useDeletePost.jsx';
+
+
+export const PostsComponent = ({ statePost, setStatePost, user, setUser }) => {
+  const [postRes, getPost] = useGetPost();
+  const [res, doLike] = useDoLike()
+  const [unlikeRes, doUnLike] = useDoUnLike()
+  const [currentPostId, setCurrentPostId] = useState()
+  const [errorMessage, setErrorMessage] = useState('');
+  const [userRes, getUser] = useGetUser()
+  const[resDel,doDel]=useDeletePost()
+
+  useEffect(() => {
+    getUser()
+  }, [])
+
+  useEffect(() => {
+    if (userRes) {
+      setUser(userRes.data)
+    }
+
+  }, [userRes])
+
+  useEffect(() => {
+    getPost()
+  }, [])
+
+  useEffect(() => {
+    console.log(postRes)
+    if (postRes.data) {
+      setStatePost(postRes.data)
+    }
+  }, [postRes])
+
+  useEffect(() => {
+    const index = statePost?.findIndex(statePost => statePost._id === currentPostId)
+    const tempPosts = [...statePost]
+    if (res.status === 200) {
+      if (tempPosts) {
```

Comment:
we could be combined both `if` conditions

Reply:
fixed

Reply:
how is this fixed? shouldn't we be using `if(res.status === 200 && tempPosts) {`